### ISSUE TITLE
Are there problems with Liquid Handling?

### ISSUE THREAD

**Issue by mdedmon** [Ipsis/Woot#511]
I now know why you say that you'll either love or hate this mod. But, with that said... I'm seeing some strange behaviors and they tend to always be with liquids. I am running this on a hosted server setup.

** Updated ** running the Direwolf 1.7.0 modpack with included the woot 1.16.4-1.0.3.2 

I'll make a short video of this and upload a link to it as well. I think this mod would be super cool to get working, but I am struggling.

Link to short video showing issues:  [https://youtu.be/uCmDtslQmac](https://youtu.be/uCmDtslQmac)

2 Issues below: 

**1)**  Built a Dye Liquifier. Built an Injection press. Filled all of the dye liquifier with flowers until it was full. The tank on the right side of the GUI began filling with Pure Dye Fluid. I Placed the Injection Press directly to the right of the liquifier and nothing happened. The liquid didn't move. I finally figured out that you needed glowstone dust to change from static to dump (still have NO idea what that means), but tried it anyway. boom, the pure dye fluid moved into the injection press.

I then put pink dye casings followed by purple dye casings in the press. The liquid drained away. But, it didn't refill.

Now, the liquifier shows completely full again (7200mb on each color), but the tank shows 0/4000mb and won't refill. I've tried switching between strict / dump, but nothing changes. I can break and place the machine and still see the same. It won't take any more dye/flowers.

So, I built a new dye liquifier and it is working. I have no idea why. But the broken machine is still unusable.

**2)**  Built a Zelator[I] - I figured out with some guessing / troubleshooting how to get this thing built. The Intern would not build the machine, even with all the items in my inventory. So, I placed them manually using the Layout Guide. Once I got all the pieces in place, they "melded" to a single unit. So, we're good there. I can access the interface by right clicking on the Heart. I couldn't figure out how to get power(?) to the unit, but after looking at 1.12 videos and this build, I guessed I would need Conatus fluid for the Conatus basic cell at the bottom of the build. I placed a fluid vat directly under the Conatus basic cell. Placed an enchantment liquifier next to that, powered them both and began liquifying enchantments. That worked for the first "round"(?). The Heart filled up with Conatus fluid and began exporting loot. But the fluid emptied out of the heart and never refilled, even though the 2 machines at the bottom are still working and the VAT shows it's full of conatus fluid, but it's not going into the conatus basic cell.

I'm sorry this is so wordy, but hope it helps.

**Comment by Ipsis** [comment]
Thanks for the video - that is always helpful.

TLDR - 1) bug, 2) setup error

1) So the static and dump is basically to stop the machine getting stuck.
In static mode, if you put a flower into the liquifier and it generates dye that the internal tanks cannot store, then it stops working - ie you need to actively manage what you put in.
In dynamic mode, as long as there is space for at least one internal dye color to be filled, then it is processed and any excess it dumped. (eg. purple -> red and blue dye, if only space for blue, then the red is dumped)

As you saw it stores the tank contents when you break it - currently I've not added any recipe to clear them in a crafting grid. However you should be able to bucket out from the front face.

The liquifier tries on every tick to find an adjacent tank to put the liquid in and side shouldn't matter.
So far I've been unable to reproduce the liquifier issue that you showed to try and debug it.

Edit - I think I've recreated it, but it only seems to be when all internal tanks are full ie. all 7200/7200/7200/7200 - I have space in the output tank but the pure dye is not being created.

So the pure dye is only created at the end of processing the input. When that happens everything that will fit in the output tank is converted. However if you have a full output tank and then fill the input tanks as well, then there is no input item to process and trigger the output tank filling. The machine is literally jammed. So that is a bug and draining the output tank should really see if any more processing of the internal can be done.

2) Intern will place all the blocks except for the last two (cell and controller)  - those need placed manually. 
The older versions of Woot did use RF (1.12) but for these versions I dumped RF and moved over to Conatus Fluid as the power mechanism. 
I recreated your setup and the VAT was successfully pushing the fluid into the cell for me however I have mine with the VAT beside the cell. The VAT will only output fluid from the bottom, back and right sides. The Woot guide should explain input and output sides for each machine.
